diff -ruN p7zip_4.27./makefile.linux_amd64 p7zip_4.27/makefile.linux_amd64
--- p7zip_4.27./makefile.linux_amd64	2005-09-22 19:54:06.000000000 +0200
+++ p7zip_4.27/makefile.linux_amd64	2005-09-23 12:02:21.880417000 +0200
@@ -1,12 +1,12 @@
 
 # use "-m64" to have a 64bits executable
-ALLFLAGS=-m32 -w -O -s  -fPIC \
+ALLFLAGS=-m32 -w -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
diff -ruN p7zip_4.27./makefile.linux_x86_ppc_alpha p7zip_4.27/makefile.linux_x86_ppc_alpha
--- p7zip_4.27./makefile.linux_x86_ppc_alpha	2005-09-22 19:54:03.000000000 +0200
+++ p7zip_4.27/makefile.linux_x86_ppc_alpha	2005-09-23 12:02:27.885788500 +0200
@@ -2,13 +2,13 @@
 # makefile for Linux (x86, PPC, alpha ...)
 # LITTLE_ENDIAN and BIG_ENDIAN are already defined by the system headers
 #
-ALLFLAGS=-O1 -s -fPIC \
+ALLFLAGS=-fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
