diff -ruN p7zip_4.39./makefile.linux_amd64 p7zip_4.39/makefile.linux_amd64
--- p7zip_4.39./makefile.linux_amd64	2006-04-15 13:16:07.000000000 +0200
+++ p7zip_4.39/makefile.linux_amd64	2006-04-15 13:16:29.481547392 +0200
@@ -1,12 +1,12 @@
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 -w -O -s  -fPIC \
+ALLFLAGS=-m64 -w -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
diff -ruN p7zip_4.39./makefile.linux_x86_ppc_alpha p7zip_4.39/makefile.linux_x86_ppc_alpha
--- p7zip_4.39./makefile.linux_x86_ppc_alpha	2006-04-15 13:16:07.000000000 +0200
+++ p7zip_4.39/makefile.linux_x86_ppc_alpha	2006-04-15 13:16:45.061178928 +0200
@@ -2,13 +2,13 @@
 # makefile for Linux (x86, PPC, alpha ...)
 # LITTLE_ENDIAN and BIG_ENDIAN are already defined by the system headers
 #
-ALLFLAGS=-O1 -s -fPIC \
+ALLFLAGS=-fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
diff -ruN p7zip_4.39./makefile.linux_x86_ppc_alpha__gcc_4.X p7zip_4.39/makefile.linux_x86_ppc_alpha__gcc_4.X
--- p7zip_4.39./makefile.linux_x86_ppc_alpha__gcc_4.X	2006-04-15 13:16:07.000000000 +0200
+++ p7zip_4.39/makefile.linux_x86_ppc_alpha__gcc_4.X	2006-04-15 13:16:58.236176024 +0200
@@ -2,14 +2,14 @@
 # makefile for Linux (x86, PPC, alpha ...)
 # LITTLE_ENDIAN and BIG_ENDIAN are already defined by the system headers
 #
-ALLFLAGS=-O1 -s -fPIC \
+ALLFLAGS=-fPIC \
 	-DHAVE_GCCVISIBILITYPATCH -fvisibility=hidden -fvisibility-inlines-hidden \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
