diff -ur p7zip_4.27/makefile.linux_amd64 p7zip_4.27.new/makefile.linux_amd64
--- p7zip_4.27/makefile.linux_amd64	2005-09-22 17:54:06.000000000 +0000
+++ p7zip_4.27.new/makefile.linux_amd64	2005-09-23 07:11:37.642811944 +0000
@@ -5,8 +5,8 @@
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
diff -ur p7zip_4.27/makefile.linux_x86_ppc_alpha p7zip_4.27.new/makefile.linux_x86_ppc_alpha
--- p7zip_4.27/makefile.linux_x86_ppc_alpha	2005-09-22 17:54:03.000000000 +0000
+++ p7zip_4.27.new/makefile.linux_x86_ppc_alpha	2005-09-23 07:12:06.162476296 +0000
@@ -7,8 +7,8 @@
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
-CC=gcc $(ALLFLAGS)
+CXX=$(_CXX) $(ALLFLAGS)
+CC=$(_CC) $(ALLFLAGS)
 LINK_SHARED=-shared
 
 LOCAL_LIBS=-lpthread
